test/server/filter_config_test.cc

```diff
@@ -0,0 +1,58 @@
+#include "envoy/server/filter_config.h"
+
+#include "test/mocks/server/factory_context.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+class TestHttpFilter : public Http::StreamDecoderFilter {
```

Comment:
Maybe extend the PassThroughDecoderFilter in extensions/filters/http/common? One less place to update if filters get a new virtual method.

Reply:
No need to extend, I just plugged the filter in. I was debating whether we'd anticipate future changes needed, but actually most of the future work here will happen at the FilterFactory level so left that stuff in place.